src/reinstall.ts

```diff
@@ -4,7 +4,6 @@ import { execFileSync } from 'child_process'
 import { safeExit } from './safeExit'
 safeExit(() => {
   console.log(`Reinstall [${process.cwd()}]`)
-  execFileSync('rm', ['-r', '-f', './node_modules'], { stdio: 'inherit' })
-  execFileSync('rm', ['-f', './yarn.lock'], { stdio: 'inherit' })
-  execFileSync('yarn', ['install'], { stdio: 'inherit' })
+  execFileSync('yarn', ['rimraf', './node_modules'], { shell: true, stdio: 'inherit' })
+  execFileSync('yarn', ['upgrade'], { shell: true, stdio: 'inherit' })
```

Comment:
@arietrouw This is the one I'm not entirely sure about.  I think the behavior we're looking for is to:

1. Remove the `yarn.lock` file
2. Run `yarn install` to recreate the `yarn.lock` file and force the dependencies to be recalculated to their latest version based on the version range specified in the package.json

From [the docs](https://classic.yarnpkg.com/lang/en/docs/cli/upgrade/) on `yarn upgrade`:

> This command updates dependencies to their latest version based on the version range specified in the package.json file. The yarn.lock file will be recreated as well.

The win when we let yarn "recreate" the file for us is that, conceptually, it's just clearing the file & recalculating the dependences, it doesn't do a `rm yarn.lock && touch yarn.lock`.  Why this difference matters is that yarn's behavior when writing the `yarn.lock` file is to:

1. First, check if the `yarn.lock` file exists already and try and infer the file's line endings (`LF` for Mac, `CR/LF` for Windows) from the file itself
2. If no `yarn.lock` file exists (like is the case when we remove it) yarn will simply use whatever line ending is specified by [os.EOL](https://nodejs.org/api/os.html#os_os_eol)

Doing it the original way means that, depending on which platform last ran this command, `yarn reinstall` will potentially change every single line of the `yarn.lock` file (even if no dependency versions changed) because every line ending will be rewritten to match the host's EOL.  When we do it this way it preserves the original line endings, no matter what platform you're on, and makes for a much more manageable diff.

Reply:
Another tool at our disposal that @jonesmac mentioned was [yarn autoclean](https://classic.yarnpkg.com/en/docs/cli/autoclean) which

> frees up space by removing unnecessary files and folders from dependencies. It reduces the number of files in your project’s node_modules folder

It might be:

1. a little more surgical/faster
2. nice to rely on idiomatic tooling if it meets our needs